frontend/app/src/App.tsx

```diff
@@ -635,6 +635,17 @@ export class App extends PureComponent<Props, State> {
     this.maybeShowErrorDialog(newDialog, errorMarkdown)
   }
 
+  showConnectionError(title: string, errorMarkdown: string): void {
+    LOG.error(errorMarkdown)
+    const newDialog: DialogProps = {
+      type: DialogType.CONNECTION_ERROR,
```

Comment:
Previously, connection errors used the `WARNING` dialog type, but for this work I needed to know whether there was a connection error dialog open, so I could rescind it. So I added a new type of dialog called `CONNECTION_ERROR`, and duplicated some of the word that handled the warning dialog.